pkg/arvo/sys/vane/jael.hoon

```diff
@@ -2416,55 +1047,50 @@
     ?:  ?=(%earl rac)
       ?.  =(u.who p.why)
         [~ ~]
-      ?:  (gth u.lyf lyf.own.sub.lex)
+      ?:  (gth u.lyf lyf.own.pki.lex)
         ~
-      ?:  (lth u.lyf lyf.own.sub.lex)
+      ?:  (lth u.lyf lyf.own.pki.lex)
         [~ ~]
-      =/  sec  (~(got by jaw.own.sub.lex) u.lyf)
+      =/  sec  (~(got by jaw.own.pki.lex) u.lyf)
       =/  cub  (nol:nu:crub:crypto sec)
       :^  ~  ~  %noun
       !>  ^-  deed:ames
-      [u.lyf pub:ex:cub sig.own.sub.lex]
+      [u.lyf pub:ex:cub sig.own.pki.lex]
     ::
-    =/  pub  (~(get by kyz.puk.sub.lex) u.who)
+    =/  pub  (~(get by pos.zim.pki.lex) u.who)
     ?~  pub
       ~
     ?:  (gth u.lyf life.u.pub)
       ~
-    =/  pas  (~(get by pubs.u.pub) u.lyf)
+    =/  pas  (~(get by keys.u.pub) u.lyf)
     ?~  pas
       ~
     :^  ~  ~  %noun
-    !>  `deed:ames`[u.lyf u.pas ~]
+    !>  `deed:ames`[u.lyf pass.u.pas ~]
   ::
       %earl
-    ?.  ?=([@ @ @ ~] tyl)  [~ ~]
-    ?.  ?&  ?=(%& -.why)
-            (~(has by pry.urb.lex) p.why)
-        ==
+    ?.  ?=([@ @ ~] tyl)  [~ ~]
+    ?.  =([%& our] why)
       [~ ~]
     =/  who  (slaw %p i.tyl)
     =/  lyf  (slaw %ud i.t.tyl)
-    =/  pub  (slaw %ux i.t.t.tyl)
     ?~  who  [~ ~]
     ?~  lyf  [~ ~]
-    ?~  pub  [~ ~]
-    ?:  (gth u.lyf lyf.own.sub.lex)
+    ?:  (gth u.lyf lyf.own.pki.lex)
       ~
-    ?:  (lth u.lyf lyf.own.sub.lex)
+    ?:  (lth u.lyf lyf.own.pki.lex)
       [~ ~]
     :: XX check that who/lyf hasn't been booted
     ::
-    =/  sec  (~(got by jaw.own.sub.lex) u.lyf)
-    =/  cub  (nol:nu:crub:crypto sec)
-    =/  sig  (sign:as:cub (shaf %earl (sham u.who u.lyf u.pub)))
-    ``[%atom !>(sig)]
+    =/  sec  (~(got by jaw.own.pki.lex) u.lyf)
+    =/  moon-sec  (shaf %earl (sham our u.lyf sec u.who))
+    =/  cub  (pit:nu:crub:crypto 128 moon-sec)
```

Comment:
Do we gain anything (other than scry convenience) by deterministically deriving the moon keys from that parent's? If peers have to ask us for the moon's public key anyway, wouldn't it be better to just statefully derive the moon's keys from entropy?

Reply:
Yes, they should be statefully derived.  Basically, there should be a `|moon` command that registers a public key with the planet's jael.  If no key is supplied, it should generate and output a private key.

This is all formally correct for now, but I'll try to build |moon soon.

Reply:
Done, PTAL at gen/hood/moon{,-breach,-cycle-keys}.hoon